rpc/gnmi/mixed-schema.md

```diff
@@ -99,6 +96,35 @@ constrained to the specified origin. Particularly:
   the specified `origin`. To delete contents from multiple origins, a client
   MUST specify multiple paths within the `delete` of the `SetRequest`.
 
+### Special considerations for the use of Origin CLI
+
+The following rules apply to a `SetRequest` containing `origin` CLI.  TODO:
+Special considerations for mixing native schema are currently undefined.
+
+* If a `SetRequest` contains `origin` `cli` and `origin` `openconfig`, the CLI
+  MUST be given precedence over the OpenConfig origin.  This precedence is
+  independent of the order in which origin CLI appears.  This overrides the
+  gNMI specification requirement of ordered path based precedence within a
+  `SetRequest`.  The goal of this precedence is to create deterministic
+  behavior for resolving overlapping configuration.  For setRequests without
+  `origin` CLI, order based precedence still applies.
+* The `update` operation MUST be supported for setRequests containing `origin`
+  `cli` and `origin` `openconfig`.  The `update` operation is compatible with
+  the requirements for `origin` `cli` to take precedence.
+* The `replace` operation for a setRequest containing only `origin` `cli` MUST
+  be treated as a full device configuration replacement.
+* The `replace` operation for a setRequest containing both CLI and OpenConfig,
```

Comment:
The wording here (and throughout) implies that implementations model each origin as it's own DB - is that really the expectation? For example if I create object `A` that exists only in `cli` and then do a replace on root in `openconfig` should object `A` still exist? https://github.com/openconfig/reference/blob/5038bb2d94444da0fe3a679ce902b8ca491b2e95/rpc/gnmi/mixed-schema.md?plain=1#L91 implies that the two shouldn't interact but it seems indirectly that they must if they are both modeling an underlying object that both can manipulate. It's not clear what the second SetRequest from the below example would look like, but would it need to continue to send the `update` portion for the `cli` origin?

To add another example, if I create object `B` in `cli` with field `F` and object `B` IS modeled in `openconfig`, without visibility to field `F`, what happens if I were to do a replace on object `B` in `openconfig`?

Now what if the path being replaced were root in both cases (implied with the `cli` origin)? Doesn't the implied replace on root in `cli` essentially make it's full contents win in a merge between the two origins?

Reply:
> The wording here (and throughout) implies that implementations model each origin as it's own DB - is that really the expectation? For example if I create object `A` that exists only in `cli` and then do a replace on root in `openconfig` should object `A` still exist?

If object A is not contained in the openconfig replace at the root, the object A should be removed.  If object A is contained in the openconfig replace at the root, object A should be replaced with the new value.
 
> https://github.com/openconfig/reference/blob/5038bb2d94444da0fe3a679ce902b8ca491b2e95/rpc/gnmi/mixed-schema.md?plain=1#L91
> 
> implies that the two shouldn't interact but it seems indirectly that they must if they are both modeling an underlying object that both can manipulate. It's not clear what the second SetRequest from the below example would look like, but would it need to continue to send the `update` portion for the `cli` origin?

Thanks for catching this.  I believe the text you've linked to here needs to be updated or removed as part of this pull request as it is in conflict with the use of mixed OpenConfig and CLI.    A setrequest containing origin OC must update/replace configuration which is also modeled by CLI.  Presumably 100% of all configuration items are modeled by CLI and OpenConfig is in reality, the subset.

> To add another example, if I create object `B` in `cli` with field `F` and object `B` IS modeled in `openconfig`, without visibility to field `F`, what happens if I were to do a replace on object `B` in `openconfig`?
> 

If the setrequest only contains origin `openconfig` and replaces object `B`, then object `B` should be replaced.

If the setrequest replace contains origin `openconfig` and origin `cli` and both reference object `B`, the `cli` reference should override or take precedence over the `openconfig` reference.  

> Now what if the path being replaced were root in both cases (implied with the `cli` origin)? Doesn't the implied replace on root in `cli` essentially make it's full contents win in a merge between the two origins?

Yes, I think this is why a setrequest with `replace` origin `openconfig` and `replace` origin `cli` in a single setrequest is not a useful operation.  The `cli` replace will take precedence rendering the `openconfig` origin data useless.   

Instead I believe the useful pattern is a setrequest with `replace` origin `openconfig` and `update` origin `cli` is in the example now.  If the `replace` origin `openconfig` contains a root path, then the full configuration is replaced with the contents of the openconfig, followed by an `update` of the origin `cli` as per [gNMI 3.4](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#34-modifying-state)